includes/delete_datastream.form.inc

```diff
@@ -28,7 +33,18 @@ function islandora_delete_datastream_form(array $form, array &$form_state, Abstr
     t('Cancel')
   );
 }
-
+/**
+ * Purge a given DSID's derivative datastreams on an object.
+ */
+function islandora_datastream_derivatives_purged(AbstractObject $object, $dsid) {
+  module_load_include('inc', 'islandora', 'includes/utilities');
+  $hooks = islandora_invoke_hook_list(ISLANDORA_DERVIATIVE_CREATION_HOOK, $object->models, array($object));
+  $dsid_map = function ($hook_definition) {
+    return $hook_definition['destination_dsid'];
+  };
+  $dsids = array_map($dsid_map, $hooks);
+  array_map(array($object, 'purgeDatastream'), $dsids);
```

Comment:
Currently, this would purge *all* derived datastreams on the object, not just those derived from the datastream being purged... Should [`filter`](https://github.com/Islandora/islandora/blob/7.x/includes/derivatives.inc#L205-L250) down to those derived from the current `$dsid`.